agentMET4FOF/agents.py

```diff
@@ -830,15 +830,23 @@ def store(self, agent_from, data=None):
                 # check if the key exist
                 # if it does, then append
                 if key in self.buffer[agent_from].keys():
-                    self.buffer[agent_from][key] = self._concatenate(self.buffer[agent_from][key], value)
+                    self.buffer[agent_from][key] = self._concatenate(self.buffer[agent_from][key], value,concat_axis)
                 # otherwise, create new entry
                 else:
                     self.buffer[agent_from].update({key: value})
         else:
             if not self.check_supported_datatype(message_data):
                 message_data = [message_data]
-            self.buffer[agent_from] = self._concatenate(self.buffer[agent_from], message_data)
+            self.buffer[agent_from] = self._concatenate(self.buffer[agent_from], message_data,concat_axis)
 
+    def values(self):
+        return self.buffer.values()
+
+    def items(self):
+        return self.buffer.items()
+
+    def keys(self):
+        return self.buffer.keys()
```

Comment:
Shall these be properties or methos? We should introduce a docstring for each of these, to have them mentioned on ReadTheDocs and in the doc preview in IDEs like PyCharm.

Reply:
i was thinking of exposing some methods as though they were dicts. hence, we could do:

`for key in agentbuffer.keys():`

instead of 

`for key in agentbuffer.buffer.keys():`

Reply:
I think, this is a really good idea. My original question was reffering to the  choice between:

```python
buffer.keys()
```

which would be more dict-like and

```python
buffer.keys
```

I think mimicking original dicts is a good idea here. Then I was thinking about some kind of inheritance either from UserDict, dict or something else and came across this [well written article about how and why to do that](https://treyhunner.com/2019/04/why-you-shouldnt-inherit-from-list-and-dict-in-python/). Your choice probably is the best one. Everything (besides and explanatory docstring to make the methods appear in the docs) is fine with this.